### PR TITLE
Add option for JVM to dump its heap on OutOfMemory

### LINKED ISSUE #4192: Add -XX:+HeapDumpOnOutOfMemoryError and related Java options to Logstash 2.0
It looks like the Logstash bin script (at least on Linux) does not add the `-XX:+HeapDumpOnOutOfMemoryError` Java option on start-up.  It would be good to add this (and `-XX:HeapDumpPath=...`) for the rare occasion where LS OOM's so we might be able to dig into the heap dump for post-mortem.

### PR DESCRIPTION
**Pull request by suyograo**
Fix #4192

### CODE REVIEW

**Comment by ph** [comment]
LGTM

**Comment by ph** [comment]
OOOPSS @suyograo maybe adding this to windows too? https://github.com/elastic/logstash/blob/master/bin/setup.bat

**Comment by ph** [comment]
Nevermind last comment, I check an old version of the file.

Windows seems to support that for some time.

**Comment by ph** [comment]
Slow start today :coffee:  :coffee:  :coffee: :coffee: 

https://github.com/elastic/logstash/blob/master/bin/setup.bat#L51 <- we should enable that lime by default too ?

**Comment by suyograo** [comment]
@ph ping. I chnaged the win file path, because `$LS_HOME/logs` does not exist

File: bin/setup.bat [hunk 1]
```diff
@@ -48,7 +48,7 @@ REM Causes the JVM to dump its heap on OutOfMemory.
 set JAVA_OPTS=%JAVA_OPTS% -XX:+HeapDumpOnOutOfMemoryError
 REM The path to the heap dump location, note directory must exists and have enough
 REM space for a full heap dump.
-REM JAVA_OPTS=%JAVA_OPTS% -XX:HeapDumpPath="$LS_HOME/logs/heapdump.hprof"
+REM JAVA_OPTS=%JAVA_OPTS% -XX:HeapDumpPath="$LS_HOME/heapdump.hprof"
```
Comment (ph) [inline bin/setup.bat]: Should be `set` not `REM` ?

**Comment by ph** [comment]
LGTM, happy heap heap heap hurray dump!

**Comment by yodog** [comment]
how do i disable this?
it's taking too much space on disk, and i'm tired of deleting the file manually.

**Pull request closed without merging**